### DISCUSSION TITLE
increase network

### DISCUSSION THREAD

**Discussion by wir19** [markqvist/Reticulum#198] [Ideas]
Please make a program to cellphone. Similar to Rattlegram https://play.google.com/store/apps/details?id=com.aicodix.rattlegram&gl=US&pli=1 or http://www.whence.com/minimodem/ for linux or other.
Simple radio (for example baofeng) no closed bandwitch (PMR in my country cb is prohibited) and only cell phone. 
Ok this is not nice sound, this is not quckly (speed) but this is extremely easy.

**Comment by markqvist** [comment]
Did you have a look at [Sideband](https://github.com/markqvist/sideband)?

The thing with pure-audio interfaces is that they kinda suck for real-world use over radios, since there is no direct control of the PTT. You need to use very slow and unreliable VOX operation of the radio, and with the huge differences in how different radios implement this, and how slow/fast they are, it is a pain to implement in any reliable way.

Not saying that it's impossible, and it would certainly be useful, but it's a *much* bigger project than most people would think, and is a lot less useful in the real world, thas most people think.

I honestly think you are *much* better off using something like Sideband and an RNode.

**Reply by wir19** [reply]
I wrote about 'interface' not programs or network. I wrote about sending packages trought simple cellphone+baofeng ham radio (or other) 
meybe You can sugest add this functionality to authors

**Reply by markqvist** [reply]
> I wrote about sending packages trought simple cellphone+baofeng ham radio (or other)

Yes, I understand that :) That is exactly what I was answering to.

It is only useful if it works reliably, and for it to work reliably, just an audio cable between radio and cellphone, it is a *very* complicated and difficult thing to do.

How do I know? I have been implementing audio-band modems in software and hardware for more than a decade ;)

It would be a cool thing to have a reliable, universal audio-based interface for Reticulum, that could run on both computers and mobile devices, I agree. Actually doing it is *fiendishly difficult*, unless you want to settle for something like 30 bits per second.

**Reply by fab1an2** [reply]
30 bits /s is ok for me. But I'm not have wire.

**Reply by markqvist** [reply]
To understand more about what is involved in this, you should probably read this thread: https://github.com/markqvist/Reticulum/discussions/11

**Reply by fab1an2** [reply]
Tell me better how to send data through baofeng. The discussion is me above my ears.

**Reply by markqvist** [reply]
The best way currently is to use a dedicated modem, like [this open source implementation](https://unsigned.io/openmodem/). It supports 1200 and 2400 bps operation over most radios, and also 300 bps operation for HF operation.

In the future, the RNode firmware will also include a dedicated audio modem in software, but it is not high priority right now.

**Reply by fab1an2** [reply]
it is expensive. very good gotenna are 2 times cheaper. https://gotennamesh.com/products/mesh?utm_source=internal-link&utm_medium=menu&utm_campaign=gotenna.com

Meybe put RMS to arduino or esp and using wifi mesh adhoc similar BATMAN https://openwrt.org/docs/guide-user/network/wifi/mesh/batman ? esp are cheap and small.

**Reply by markqvist** [reply]
Dude, you really gotta take a better look around at the entire selection of available tools here ;)

You asked what the best way to send data through a handheld radio was, and I gave you the answer. Then you start talking about gotennas. You can't use your "gotenna" to shove data through your baofeng, mate ;)

If you want a dedicated radio device, why don't you just make an [RNode](https://unsigned.io/rnode/)? It's four times cheaper than that gotenna, and much more capable.

**Reply by fab1an2** [reply]
How I can create RNode. 4 times cheaper? I see 129 euro + shipping. this one device is 4/5 price of 2 gotenna.

Can You tell me how make it 4 times cheaper?  Can You show any video ? Tutorial , how using RNode in real situations. For example in town

**Reply by markqvist** [reply]
Did you actually *read* the page? It is full of these things called "links". Great thing, those links.

When you click/tap on them, you are presented with new pages, full of information.

Since that is apparently a bit too much to ask, I helped out by copying some of the links from that page, and pasted them here:

- Create a [basic RNode from readily available development boards](https://unsigned.io/guides/2022_01_25_installing-rnode-firmware-on-supported-devices.html)
- Follow a complete build recipe for [making a handheld RNode](https://unsigned.io/guides/2023_01_14_Making_A_Handheld_RNode.html), like the one pictured above
- Learn the basics on how to [create and build your own RNode designs](https://unsigned.io/guides/2022_01_26_how-to-make-your-own-rnodes.html) from scratch

I will be happy to answer any questions and help you in finding answers you need, but please do take the effort of at least *superficially skimming* the information made available and published literally all over the place before acting like an entitled customer. Otherwise, you will just be wasting everyones time.

**Reply by fab1an2** [reply]
I did not find any references that rnode works on other devices as well. I have looked at your site several times and have not found this information anywhere. I was convinced that RNode is only an expensive device available from your site.

**Reply by markqvist** [reply]
Third menu item, one click from the front page:

https://unsigned.io/guides/

Great place to start :)

**Reply by buhalo1** [reply]
> Tell me better how to send data through baofeng. The discussion is me above my ears.

It would be better to have a separate audio modem app for android that works with a cm108 usb sound fob with a gpio ptt interface and supports the kiss ip interface. Mobile version of direwolf soundmodem.

Currently, for networking with baofengs, you can use direwolf soundmodem running on rpi zero w.

Alternatively, a minimodem or an open modem or a Mobilinkd TNC4 can currently be used.

Reticulum supports the kiss interface required to work with these modems, sideband probably does not.
In the case of sideband, it is likely that the phone can communicate with rpi0w directly over the reticulum network.

Although, of course, soundmodem support via cm108 sound fob built directly into sideband would be much more preferable, since it significantly reduces the entry threshold for users who are unable to deploy more complex network configurations.

**Reply by wir19** [reply]
> Did you have a look at Sideband

I tried to install , but I did not succeed. I would prefer to have this program in a normal store.

I use it and this is working for me. This is not perfect but this working better than sending my voice over the radio. I'd rather have something that works than nothing. I think a joint program would be a great way to increase network coverage when there is no other communication.

**Comment by srem1** [comment]
In my opinion AndFlMsg is better than rattlegram. https://sourceforge.net/projects/fldigi/files/AndFlmsg/
https://www.youtube.com/watch?v=nvu3oasPmKU&ab_channel=SnowLynx 

More protocols increase speed and error correct trought transport radio.

**Reply by buhalo1** [reply]
I am annoyed by the overload of fldigi and similar amateur programs, I want something simpler. The use of reticulum on hf bands is a priority for me, at the moment I am trying to come up with an ideal configuration for such a network, shortwave communication has its own specifics since the same node must be able to operate at different frequencies to reach different nodes, and other nodes need to know at what frequency they can reach my node. Unfortunately no existing solution allows design the network optimally. Now I mainly look towards STANAG 4415, STANAG 4285, STANAG 4538, STANAG 4539, MIL-STD-188-141A, MIL-
STD-188-141B. I hope to create a separate topic in the near future.
